samcli/local/lambdafn/runtime.py

```diff
@@ -59,62 +222,52 @@ def invoke(self, function_config, event, debug_context=None, stdout=None, stderr
-                #
-                # Start the timer **after** container starts. Container startup takes several seconds, only after which,
-                # our Lambda function code will run. Starting the timer is a reasonable approximation that function has
-                # started running.
-                timer = self._configure_interrupt(
-                    function_config.name, function_config.timeout, container, bool(debug_context)
-                )
-
-                # NOTE: BLOCKING METHOD
-                # Block the thread waiting to fetch logs from the container. This method will return after container
-                # terminates, either successfully or killed by one of the interrupt handlers above.
-                if extensions_preview_enabled():
-                    container.wait_for_result(name=function_config.name, event=event, stdout=stdout, stderr=stderr)
-                else:
-                    container.wait_for_logs(stdout=stdout, stderr=stderr)
-
-            except KeyboardInterrupt:
-                # When user presses Ctrl+C, we receive a Keyboard Interrupt. This is especially very common when
-                # container is in debugging mode. We have special handling of Ctrl+C. So handle KeyboardInterrupt
-                # and swallow the exception. The ``finally`` block will also take care of cleaning it up.
-                LOG.debug("Ctrl+C was pressed. Aborting Lambda execution")
-
-            finally:
-                # We will be done with execution, if either the execution completed or an interrupt was fired
-                # Any case, cleanup the timer and container.
-                #
-                # If we are in debugging mode, timer would not be created. So skip cleanup of the timer
-                if timer:
-                    timer.cancel()
+        except KeyboardInterrupt:
+            # When user presses Ctrl+C, we receive a Keyboard Interrupt. This is especially very common when
+            # container is in debugging mode. We have special handling of Ctrl+C. So handle KeyboardInterrupt
+            # and swallow the exception. The ``finally`` block will also take care of cleaning it up.
+            LOG.debug("Ctrl+C was pressed. Aborting Lambda execution")
+
+        finally:
+            # We will be done with execution, if either the execution completed or an interrupt was fired
+            # Any case, cleanup the timer and container.
+            #
+            # If we are in debugging mode, timer would not be created. So skip cleanup of the timer
+            if timer:
+                timer.cancel()
+            if container and not self._warm_containers:
                 self._container_manager.stop(container)
 
+    def get_container(self, function_name):
+        """
+        return the created container for the passed function_name.
+
+        :param str function_name: lambda function name
+        """
+        return self._containers.get(function_name, None)
```

Comment:
We support multiple invokes. 
We just cache the containers in case is warm containers is enabled, and for invoke .. we send http request to the RAPID server, and since we can send multiple HTTP requests to the RAPID server, so we can do multiple invocation.